src/fields/extend_p.rs

```diff
@@ -91,34 +118,53 @@ where
     let d = nodes.len() - 1;
 
     // Precompute (-1)^i * (d choose i) * p(i).
-    let convolution_right_terms = nodes
+    let mut convolution_right_terms = Vec::with_capacity(context.ntt_size);
+    convolution_right_terms.extend(
+        nodes
+            .iter()
+            .enumerate()
+            .zip(context.binomial_coefficients.iter())
```

Comment:
nittiest of nits: `context.binomial_coefficients` is `IntoIter` so we can drop the `iter()` call?